ReportingServicesTools/Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -202,13 +234,44 @@ function Set-RsDatabase
             Write-Verbose "Executing database creation script..."
             try
             {
+                $parameters = @{
+                    ServerInstance = $DatabaseServerName
+                    Query = $SQLScript
+                    QueryTimeout = $QueryTimeout
+                    ErrorAction = "Stop"
+                }
+
                 if ($isSQLAdminAccount)
                 {
-                    Invoke-Sqlcmd -ServerInstance $DatabaseServerName -Query $SQLScript -QueryTimeout $QueryTimeout -ErrorAction Stop -Username $adminUsername -Password $adminPassword
+                    $parameters.add("Username", $adminUsername)
+                    $parameters.add("Password", $adminPassword)
+                }
+
+                if ($containsSQLServerV22Parameters)
```

Comment:
this code is duplicated twice in this cmdlet consider moving up so its only done once

Reply:
Moved Invoke-Sqlcmd general parameters composition to a separated region.